connstr.go

```diff
@@ -95,6 +115,10 @@ func Parse(connStr string) (out ConnSpec, err error) {
 		case "couchbase":
 		case "couchbases":
 		case "http":
+		case "ns_server":
+			onlyAllowSingleHost = true
```

Comment:
I think we should move this scheme specific behaviour up a level to the specific application, or have it as part of resolution.